### ISSUE TITLE
libvirt Service Name Incorrect on Ubuntu 16.10

### ISSUE THREAD

**Issue by haraldrudell** [saltstack-formulas/libvirt-formula#20]
libvirt/map.jinja has it that

> 'libvirt_service': 'libvirtd' if grains.get('osmajorrelease', 0)|float >= 8 else 'libvirt-bin',

However, checking the grains:
```
sudo salt [minionid] grains.items | less
…
   os:
        Ubuntu
    os_family:
        Debian
    osarch:
        amd64
    oscodename:
        yakkety
    osfinger:
        Ubuntu-16.10
    osfullname:
        Ubuntu
    osrelease:
        16.10
    osrelease_info:
        - 16
        - 10
```

so the unit name used is libvirt-bin, where the correct name is **libvirtd**
This produces the error
>           ID: libvirt.service
>     Function: service.running
>         Name: libvirt-bin
>       Result: False
>      Comment: Failed to get information on unit 'libvirt-bin'

Verifying on the minion:
```
systemctl status libvirtd.service 
● libvirtd.service - Virtualization daemon
```
**get-around**
edit map.jinja and make the line 11:
'libvirt_service': 'libvirtd',

**Comment by iggy** [comment]
This sounds more like a Salt bug that osmajorrelease isn't set in 16.10... In 16.04 it is definitely set to 16.

What version of salt are you running?

**Comment by haraldrudell** [comment]
both minions 2015.8.8 and 2016.3.1 do not have

**Comment by haraldrudell** [comment]
I upgraded to 2016.11.0 where it appears to be working again:

```
sudo salt c89 grains.items | grep --after-context=1 osmajorrelease:
    osmajorrelease:
        16
```

**Comment by ukretschmer** [comment]
So the question is: Should the formula support the older buggy salt versions, which don't have the osmajorrelease grain? I guess not and therefor this issue could be closed.

For those running into trouble caused by the missing majorrelease information the workaround is to set the correct service name in Pillar data and overwrite the defaults from map.jinja:

```
libvirt:
  lookup:
    libvirt_service: libvirtd
```

**Comment by iggy** [comment]
To be specific, those minion versions are only missing that grain for that specific version of Ubuntu. That grain does exist for other versions of Ubuntu. Thanks for getting us the info and thanks for the simple workaround ukretschmer.

We should probably leave this issue open for a while just so it's easier to find (nobody ever searches closed bugs).

**Comment by baby-gnu** [comment]
Hello.

Could it be closed now since it's 2 years old and 2016.3 is [unsupported now](https://repo.saltstack.com/2016.3.html)?

Regards.
